pywavefront/obj.py

```diff
@@ -0,0 +1,205 @@
+
+    @auto_consume
+    def parse_o(self):
+        self.mesh = Mesh(self.values[1])
+        self.wavefront.add_mesh(self.mesh)
+
+    def parse_f(self):
+        # Support objects without `o` statement
+        if self.mesh is None:
+            self.mesh = Mesh()
+            self.wavefront.add_mesh(self.mesh)
+
+        # Add default material if not created
+        if self.material is None:
+            self.material = Material(is_default=True)
+            self.wavefront.materials[self.material.name] = self.material
+
+        self.mesh.add_material(self.material)
+
+        self.material.vertices += list(self.consume_faces())
+
+        # Since list() also consumes StopIteration we need to sanity check the line
+        # to make sure the parser advances
+        if self.values[0] == "f":
+            self.next_line()
+
+    def consume_faces(self):
+        """
+        Consume all consecutive faces
+
+        If a 4th vertex is specified, we triangulate.
+        In a perfect world we could consume this straight forward and draw using GL_TRIANGLE_FAN.
+        This is however rarely the case..
+
+        * If the face is co-planar but concave, then you need to triangulate the face
+        * If the face is not-coplanar, you are screwed, because OBJ doesn't preserve enough information
+          to know what tessellation was intended
+
+        We always triangulate to make it simple
+        """
+        while True:
+            v1 = None
+            vlast = None
+
+            for i, v in enumerate(self.values[1:]):
+                v_index, t_index, n_index = (list(map(int, [j or 0 for j in v.split('/')])) + [0, 0])[:3]
+
+                # Resolve negative index lookups
+                if v_index < 0:
+                    v_index += len(self.vertices) - 1
+
+                if t_index < 0:
+                    t_index += len(self.tex_coords) - 1
+
+                if n_index < 0:
+                    n_index += len(self.normals) - 1
+
+                vertex = (
+                    self.tex_coords[t_index][0],
+                    self.tex_coords[t_index][1],
+
+                    self.normals[n_index][0],
+                    self.normals[n_index][1],
+                    self.normals[n_index][2],
+
+                    self.vertices[v_index][0],
+                    self.vertices[v_index][1],
+                    self.vertices[v_index][2],
+                )
+
+                for v in vertex:
+                    yield v
+
+                if i >= 3:
+                    # Emit vertex 1 and 3 triangulating when a 4th vertex is specified
+                    for v in v1:
+                        yield v
+
+                    for v in vlast:
+                        yield v
+
+                if i == 0:
+                    v1 = vertex
+
+                vlast = vertex
+
+            # Break out of the loop when there are no more f statements
+            self.next_line()
+            if self.values[0] != "f":
```

Comment:
If possible, please use the `self.values[0]` condition in place of `while True`.

If that is not possible or inadvisable, please add a comment to that effect to the `while True`.

Reply:
Awesome, thanks!